pkg/config/shared.go

```diff
@@ -251,19 +259,19 @@ type InteropClient struct {
 }
 
 // IsZero returns whether conf is empty.
-func (conf InteropClient) IsZero() bool {
-	return conf == (InteropClient{})
+func (c InteropClient) IsZero() bool {
+	return c == (InteropClient{})
 }
 
 // Fetcher returns fetch.Interface defined by conf.
-func (conf InteropClient) Fetcher() fetch.Interface {
+func (c InteropClient) Fetcher() (fetch.Interface, error) {
 	switch {
-	case conf.Directory != "":
-		return fetch.FromFilesystem(conf.Directory)
-	case conf.URL != "":
-		return fetch.FromHTTP(conf.URL, true)
+	case c.Directory != "":
+		return fetch.WithBasePath(fetch.FromFilesystem(), c.Directory), nil
```

Comment:
My concern is here.

Assuming `c.Directory` is `/path/to/interop`, then calling `Fetch("/run", "secrets", "foo")` takes `basepath.goL33`, and the `fsFetcher` happily returns the foo secret, because `f.root` is empty.

It's way too easy to build insecure implementations with this.

Reply:
1. Creating a fetcher with empty root path today is allowed and is affected by this. The PR does not add anything new there.
2. This exact fetcher is used to fetch sensitive data. The concerns should not be *what* it fetches, but *where* it exposes it.
3. The possible security risk is if e.g. the fetcher would be exposed to the network and take arbitrary paths from the network and either expose the contents or other details(e.g. error - whether a file exists or not), however we do not do that and if we did, any issues would be mitigated by simply setting the root path.

Overall, I don't see anything to be concerned about here. The user configures the path to read here and runs the process. There's no third party anyhow affecting the paths read and/or being able to access the contents.
For extra safety the process can be run in an isolated environment, where no unnecessary sensitive data is available. In any case, nothing like this belongs here.

Reply:
1. Yes, but then this is done with explicit config so people know what they're getting into.
2. Sure, but I still think that should be limited to the configured root boundary. If a directory is configured as root, that shouldn't just be ignored.
3. _"any issues would be mitigated by simply setting the root path"_  
  that is exactly what I'm asking.

Bumping this over to @johanstokking.

Reply:
So the purpose of all of this is to be able to fetch arbitrary absolute paths.
That is not possible with root of fetcher being anything else than `/` or ``.
We need to be able to fetch arbitrary absolute paths - we discussed this on Slack as well. It seems that you're against that.

What other solution did you have in mind? Can you give any kind of other suggestion, which can be applied here?